src/main/java/org/cryptomator/ui/mainwindow/ResizeController.java

```diff
@@ -53,48 +53,70 @@ public class ResizeController implements FxController {
 	public void initialize() {
 		LOG.trace("init ResizeController");
 
-		if (neverTouched()) {
-			settings.displayConfiguration.set(getMonitorSizes());
-			return;
-		} else {
-			if (didDisplayConfigurationChange()) {
-				//If the position is illegal, then the window appears on the main screen in the middle of the window.
-				Rectangle2D primaryScreenBounds = Screen.getPrimary().getBounds();
-				window.setX((primaryScreenBounds.getWidth() - window.getMinWidth()) / 2);
-				window.setY((primaryScreenBounds.getHeight() - window.getMinHeight()) / 2);
-				window.setWidth(window.getMinWidth());
-				window.setHeight(window.getMinHeight());
-			} else {
-				window.setHeight(settings.windowHeight.get() > window.getMinHeight() ? settings.windowHeight.get() : window.getMinHeight());
-				window.setWidth(settings.windowWidth.get() > window.getMinWidth() ? settings.windowWidth.get() : window.getMinWidth());
-				window.setX(settings.windowXPosition.get());
-				window.setY(settings.windowYPosition.get());
-			}
+		if (!neverTouched()) {
+			window.setHeight(settings.windowHeight.get() > window.getMinHeight() ? settings.windowHeight.get() : window.getMinHeight());
+			window.setWidth(settings.windowWidth.get() > window.getMinWidth() ? settings.windowWidth.get() : window.getMinWidth());
+			window.setX(settings.windowXPosition.get());
+			window.setY(settings.windowYPosition.get());
 		}
-		savePositionalSettings();
+
+		window.setOnShowing(this::checkDisplayBounds);
 	}
 
 	private boolean neverTouched() {
 		return (settings.windowHeight.get() == 0) && (settings.windowWidth.get() == 0) && (settings.windowXPosition.get() == 0) && (settings.windowYPosition.get() == 0);
 	}
 
-	private boolean didDisplayConfigurationChange() {
-		String currentDisplayConfiguration = getMonitorSizes();
-		String settingsDisplayConfiguration = settings.displayConfiguration.get();
-		boolean configurationHasChanged = !settingsDisplayConfiguration.equals(currentDisplayConfiguration);
-		if (configurationHasChanged) settings.displayConfiguration.set(currentDisplayConfiguration);
-		return configurationHasChanged;
+	private boolean isWithinDisplayBounds() {
+		// (x1, y1) is the top left corner of the window, (x2, y2) is the bottom right corner
+		final double slack = 10;
+		final double width = window.getWidth() - slack;
+		final double height = window.getHeight() - slack;
+		final double x1 = window.getX() + slack;
+		final double y1 = window.getY() + slack;
+		final double x2 = x1 + width;
+		final double y2 = y1 + height;
```

Comment:
You must substract the slack twice. Otherwise the right bottom corner is the one of the original rectangle, as can be seen mathematically:
```
x2 = x1 + width = (window.getX() + slack) + (window.getWidth() - slack) = window.getX() + window.getWidth()
```

Reply:
Ah, yes of course.